src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -120,19 +106,130 @@ def put(self, args):
     def get(self):
         return self.output_queue.get(timeout=1)
 
-    def format_current_process(self, line_number: int, is_completed=False):
-        process_name = self.process.name if self.process else None
-        process_pid = self.process.pid if self.process else None
-        if is_completed:
-            bulk_logger.info(
-                f"Process name: {process_name}, Process id: {process_pid}, Line number: {line_number} completed."
-            )
-        else:
-            bulk_logger.info(
-                f"Process name: {process_name}, Process id: {process_pid}, Line number: {line_number} start execution."
-            )
 
-        return f"Process name({process_name})-Process id({process_pid})-Line number({line_number})"
+def format_current_process(process_name, pid, line_number: int, is_completed=False):
+    if is_completed:
+        bulk_logger.info(
+            f"Process name: {process_name}, Process id: {pid}, Line number: {line_number} completed."
+        )
+    else:
+        bulk_logger.info(
+            f"Process name: {process_name}, Process id: {pid}, Line number: {line_number} start execution."
+        )
+
+    return f"Process name({process_name})-Process id({pid})-Line number({line_number})"
+
+
+def fork_processes_manager(
+        log_context_initialization_func,
+        current_operation_context,
+        input_queues,
+        output_queues,
+        control_signal_queue,
+        flow_file,
+        connections,
+        working_dir,
+        raise_ex,
+):
+    signal.signal(signal.SIGINT, signal_handler)
+    process_info = {}
+    context = multiprocessing.get_context("fork")
+    run_storage = QueueRunStorage(output_queues[0])
+    executor = FlowExecutor.create(
+        flow_file=flow_file,
+        connections=connections,
+        working_dir=working_dir,
+        raise_ex=raise_ex,
+        storage=run_storage
+    )
+    executor_creation_func = partial(create_executor_fork, flow_executor=executor)
+
+    def new_process(i):
+        process = context.Process(
+            target=_process_wrapper,
+            args=(
+                executor_creation_func,
+                input_queues[i],
+                output_queues[i],
+                log_context_initialization_func,
+                current_operation_context
+            ),
+            daemon=True
+        )
+        process.start()
+        input_queues[i].put((i, process.pid, process.name))
```

Comment:
We can see that it's not obvious what does "input_queues" do and why we put a tuple (i, process.pid, procees.name) to it, please add some annotation to explain the purpose of input/output queues or use a longer name to show it's real purpose

Reply:
Get (i, process.pid, procees.name) through process_info dict, no longer from input_queue